### DISCUSSION TITLE
Same experiment output plot for different experiment script

### DISCUSSION THREAD

**Discussion by rifat37dlr** [pybamm-team/PyBaMM#3851] [Q&A]
Why do the experiments results give the same outputs even if I change the temperature dependency. For example, in code (1) I computed the experiment without any temperature dependency (Ambient temperature 298.15 K) but in code (2) I computed the experiment with temperature dependency (I used 273.15K) in step 1, 2, 5 and 6 but I got the same output as before when I did not consider any temperature dependency. I also changed the temperature to 350K, where the cell should not work due to thermal runway but the output plot was also same.

![image](https://github.com/pybamm-team/PyBaMM/assets/161710860/2e0eae7c-ae26-4fc6-95dd-835a86ea8ba3)

**Comment by rifat37dlr** [comment]
Pybamm team, could anyone help me to understand the problem?

@tinosulzer , @brosaplanella , @TomTranter , @DrSOKane , @rtimms

**Reply by rtimms** [reply]
The output doesn't look the same to me, can you plot both voltage traces on the same figure? The changes might be small as many of the parameters in the Chen set don't have temperature dependence, so the temperature is mostly coming in through Butler-Volmer. At C/30 you probably won't see much difference.

**Comment by TomTranter** [comment]
First off you need to pass a thermal model option into the DFN model. Secondly instead of pasting pictures of code can you please past code snippets

 there's a button at the top of the message box for this <>

**Reply by rifat37dlr** [reply]
Hello Tom,
Thanks for your quick response. Here is the temperature dependency experiment code: 
```
import pybamm
import matplotlib.pyplot as plt
import numpy as np

pybamm.set_logging_level('NOTICE')

options = {"thermal": "lumped"}
pram = pybamm.ParameterValues('Chen2020')
model = pybamm.lithium_ion.DFN(options)
sim = pybamm.Simulation(model)
sol = sim.solution
sim.solve([0, 36000])
voltage = sim.solution['Terminal voltage [V]'].entries

Step_1 = pybamm.step.string("Rest for 2 hours", temperature=273.15)
Step_2 = pybamm.step.string("Discharge at C/30 until 2.5 V", temperature=273.15)
Step_3 = pybamm.step.string("Rest for 2 hours", temperature=298.15)

if voltage.any() < 2.5:
    Step_4 = pybamm.step.string("Discharge at C/30 until 2.5 V", temperature=298.15)
else:
    Step_4 = pybamm.step.string("Charge at C/30 until 2.5 V", temperature=298.15)

Step_5 = pybamm.step.string("Rest for 2 hours", temperature=273.15)
Step_6 = pybamm.step.string("Charge at C/30 until 4.2 V", temperature=273.15)

Step_7 = pybamm.step.string("Rest for 2 hours", temperature=298.15)

if voltage.any() > 4.2:
    Step_8 = pybamm.step.string("Discharge at C/30 until 4.2 V", temperature=298.15)
else:
    Step_8 = pybamm.step.string("Charge at C/30 until 4.2 V", temperature=298.15)

experiment = pybamm.Experiment([Step_1, Step_2, Step_3, Step_4, Step_5, Step_6, Step_7, Step_8])


model = pybamm.lithium_ion.DFN()
sim = pybamm.Simulation(model, experiment=experiment)
sim.solve()
sim.plot(['Terminal voltage [V]', 'Current [A]'])
```

Here is the other one:
```
import pybamm
import matplotlib.pyplot as plt
import numpy as np

# pybamm.set_logging_level('NOTICE')

options = {"thermal": "lumped"}
pram = pybamm.ParameterValues('Chen2020')
model = pybamm.lithium_ion.DFN(options)
sim = pybamm.Simulation(model)
sol = sim.solution
sim.solve([0, 36000])
voltage = sim.solution['Terminal voltage [V]'].entries



Step_1 = pybamm.step.string("Rest for 2 hours")
Step_2 = pybamm.step.string("Discharge at C/30 until 2.5 V")
Step_3 = pybamm.step.string("Rest for 2 hours")

if voltage.any() < 2.5:
    Step_4 = pybamm.step.string("Discharge at C/30 until 2.5 V")
else:
    Step_4 = pybamm.step.string("Charge at C/30 until 2.5 V")

Step_5 = pybamm.step.string("Rest for 2 hours")
Step_6 = pybamm.step.string("Charge at C/30 until 4.2 V")
Step_7 = pybamm.step.string("Rest for 2 hours")

if voltage.any() < 4.2:
    Step_8 = pybamm.step.string("Charge at C/30 until 4.2 V")
else:
    Step_8 = pybamm.step.string("Discharge at C/30 until 4.2 V")

experiment = pybamm.Experiment([Step_1, Step_2, Step_3, Step_4, Step_5, Step_6, Step_7, Step_8])


model = pybamm.lithium_ion.DFN()
sim = pybamm.Simulation(model, experiment=experiment)

sim.solve()
sim.plot(['Terminal voltage [V]', 'Current [A]'])
```

**Comment by valentinsulzer** [comment]
- When opening an issue / discussion try to simplify the problem down to its most basic component (minimum working exmaple) to really isolate the problem. In this case you just need a single discharge, not a rest-discharge-rest-charge
- I don't know why you're doing this
```python
if voltage.any() < 2.5:
    Step_4 = pybamm.step.string("Discharge at C/30 until 2.5 V")
else:
    Step_4 = pybamm.step.string("Charge at C/30 until 2.5 V")
```
but you don't need to, and since you're not redefining the voltage it probably isn't doing what you think it's doing
- Plot the results on the same axes to see the differences
- PyBaMM doesn't currently have any models for thermal runaway
- Don't forget to pass the parameter values into the simulation

At C/30 the only difference is in the open circuit potential which doesn't depend strongly on temperature so there isn't much difference:
<img width="792" alt="Screenshot 2024-02-29 at 10 49 14 AM" src="https://github.com/pybamm-team/PyBaMM/assets/20817509/2366f966-68c0-4682-9f3c-88cd7b56ee02">

At 1C you can see more of a difference as the kinetics and transport depend more strongly on temperature
<img width="792" alt="Screenshot 2024-02-29 at 10 48 41 AM" src="https://github.com/pybamm-team/PyBaMM/assets/20817509/2f0c48b6-4824-460d-b12f-a33ddb98f6d2">

```python
import pybamm

param = pybamm.ParameterValues("Chen2020")
model = pybamm.lithium_ion.DFN()

sols = {}
for T in [273, 298]:
    Step_1 = pybamm.step.string("Discharge at C/30 until 2.5 V", temperature=T)

    experiment = pybamm.Experiment([Step_1])

    model = pybamm.lithium_ion.DFN()
    sim = pybamm.Simulation(model, experiment=experiment, parameter_values=param)
    sols[T] = sim.solve()
pybamm.dynamic_plot(
    list(sols.values()),
    ["Terminal voltage [V]", "Current [A]"],
    labels=["273 K", "298 K"],
)
```
